@here/harp-mapview/lib/PathBlockingElement.ts

```diff
@@ -15,5 +18,19 @@
  */
 
 export class PathBlockingElement {
-    constructor(readonly points: THREE.Vector3[]) {}
+    /**
+     * Note, lines is only used as a performance improvement and contains no useful information.
+     */
+    readonly lines: THREE.Line3[];
+
+    /**
+     * Constructs a path from a list of points.
+     * @param points Points in world coordinates.
+     */
+    constructor(readonly points: Vector3Like[]) {
+        this.lines = new Array<THREE.Line3>(points.length >= 2 ? points.length - 1 : 0);
+        for (let i = 0; i < this.lines.length; i++) {
+            this.lines[i] = new THREE.Line3(new THREE.Vector3(), new THREE.Vector3());
```

Comment:
Would be better to initialize the Line segments with the proper points. I know the purpose here is to just allocate the objects but somebody might use that class for sth. else and it would be a surprise that all line segments are 0

Reply:
These lines are in screen space, so it can contain no useful information yet. I will add a comment to clarify this.

Reply:
Check: 548a4ca2cb7575b090ecd2de5e1d0e86369ff728